### PR TITLE
1489 comments module

### PR DESCRIPTION
**Pull request by ledsoft**
https://kbss.felk.cvut.cz/redmine/issues/1489

### CODE REVIEW

File: src/main/java/cz/cvut/kbss/termit/model/comment/Comment.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import java.lang.reflect.Field;
+import java.net.URI;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents a single comment in the comment/discussion module.
+ */
+@OWLClass(iri = "http://rdfs.org/sioc/types#Comment")
```
Comment (psiotwo) [inline src/main/java/cz/cvut/kbss/termit/model/comment/Comment.java]: Constants would be better.

File: src/main/java/cz/cvut/kbss/termit/persistence/dao/comment/CommentDao.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    @Autowired
+    public CommentDao(EntityManager em, DescriptorFactory descriptorFactory, Configuration config) {
+        this.em = em;
+        this.loadingDescriptor = createLoadingDescriptor(config.get(ConfigParam.COMMENTS_CONTEXT), descriptorFactory);
+        this.savingDescriptor = createSavingDescriptor(config.get(ConfigParam.COMMENTS_CONTEXT), descriptorFactory);
+    }
+
+    private Descriptor createLoadingDescriptor(String context, DescriptorFactory descriptorFactory) {
+        final EntityDescriptor descriptor = new EntityDescriptor(URI.create(context), false);
+        descriptor.addAttributeContext(descriptorFactory.fieldSpec(Comment.class, "author"), null);
```
Comment (psiotwo) [inline src/main/java/cz/cvut/kbss/termit/persistence/dao/comment/CommentDao.java:38]: author is also inferred?

File: src/main/java/cz/cvut/kbss/termit/model/comment/CommentReaction.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.net.URI;
+import java.util.Objects;
+import java.util.Set;
+
+@OWLClass(iri = Vocabulary.s_c_reakce)
+public class CommentReaction extends AbstractEntity implements HasTypes {
+
+    @ParticipationConstraints(nonEmpty = true)
+    @OWLObjectProperty(iri = "https://www.w3.org/ns/activitystreams#actor")
```
Comment (psiotwo) [inline src/main/java/cz/cvut/kbss/termit/model/comment/CommentReaction.java]: Again, a constant?

File: src/test/java/cz/cvut/kbss/termit/persistence/dao/TermAssignmentDaoTest.java [hunk 4]
```diff
@@ -254,8 +254,8 @@ void getAssignmentsInfoByResourceRetrievesInformationAboutAssignmentsAndOccurren
             assertEquals(term.getPrimaryLabel(), rta.getTermLabel());
         });
         final Optional<ResourceTermAssignments> occ = result.stream()
-                                                            .filter(rta -> rta instanceof ResourceTermOccurrences)
-                                                            .findAny();
+                .filter(rta -> rta instanceof ResourceTermOccurrences)
```
Comment (psiotwo) [inline src/test/java/cz/cvut/kbss/termit/persistence/dao/TermAssignmentDaoTest.java]: why not `ResourceTermOccurrences.class::isInstance` ?

**Review by psiotwo — DISMISSED** [review]
lgtm, approving, minor feedback inline.

File: src/main/java/cz/cvut/kbss/termit/persistence/dao/comment/CommentDao.java (hunk 2 above)
Comment (ledsoft) [reply src/main/java/cz/cvut/kbss/termit/persistence/dao/comment/CommentDao.java:38]: Author is not inferred, but is in the default context.

File: src/test/java/cz/cvut/kbss/termit/persistence/dao/TermAssignmentDaoTest.java (hunk 4 above)
Comment (ledsoft) [reply src/test/java/cz/cvut/kbss/termit/persistence/dao/TermAssignmentDaoTest.java]: What difference does it make?

File: src/main/java/cz/cvut/kbss/termit/model/comment/Comment.java (hunk 1 above)
Comment (ledsoft) [reply src/main/java/cz/cvut/kbss/termit/model/comment/Comment.java]: Where should I put such constants?

File: src/test/java/cz/cvut/kbss/termit/persistence/dao/TermAssignmentDaoTest.java (hunk 4 above)
Comment (psiotwo) [reply src/test/java/cz/cvut/kbss/termit/persistence/dao/TermAssignmentDaoTest.java]: Not much, just a variable-free syntax only. 
Feel free to ignore.

File: src/main/java/cz/cvut/kbss/termit/model/comment/Comment.java (hunk 1 above)
Comment (psiotwo) [reply src/main/java/cz/cvut/kbss/termit/model/comment/Comment.java]: Simply to Constants.java? You can create a custom class of course, or even declare them in the termit ontology.

Comment (ledsoft) [reply src/main/java/cz/cvut/kbss/termit/model/comment/Comment.java]: I'm considering adding import of SIOC and ActivityStreams into the TermIt glossary/vocabulary. That would solve it. What do you think?

Comment (psiotwo) [reply src/main/java/cz/cvut/kbss/termit/model/comment/Comment.java]: Best option IMO.

**Review by psiotwo — APPROVED** [review]
lgtm

**Pull request merged**